WikiClientLibrary/Generators/MyWatchlistGenerator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections.Generic;
+using WikiClientLibrary.Generators.Primitive;
+using WikiClientLibrary.Infrastructures;
+using WikiClientLibrary.Sites;
+
+namespace WikiClientLibrary.Generators
+{
+    /// <summary>
+    /// Get all pages on the current user's watchlist.
+    /// </summary>
+    public class MyWatchlistGenerator : WikiPageGenerator
+    {
+        public MyWatchlistGenerator(WikiSite site) : base(site)
+        {
+        }
+
+        /// <summary>
+        /// Only list pages in the given namespaces.
+        /// </summary>
+        public IEnumerable<int>? NamespaceIds { get; set; }
+
+        /// <summary>
+        /// Adds timestamp of when the user was last notified about the edit.
+        /// </summary>
+        public bool ShowChangedTime { get; set; }
+
+        /// <summary>
+        /// Only show pages that have not been changed.
+        /// </summary>
+        public PropertyFilterOption NotChangedPagesFilter { get; set; }
+
+        /// <summary>
+        /// Gets/sets a value that indicates whether the links should be listed in
+        /// the descending order. (MediaWiki 1.19+)
+        /// </summary>
+        public bool OrderDescending { get; set; }
+
+        /// <summary>
+        /// Title (with namespace prefix) to begin enumerating from.
+        /// </summary>
+        public string? FromTitle { get; set; }
+
+        /// <summary>
+        /// Title (with namespace prefix) to stop enumerating at.
+        /// </summary>
+        public string? ToTitle { get; set; }
+
+        public override IEnumerable<KeyValuePair<string, object?>> EnumListParameters()
+        {
+            return new Dictionary<string, object?>
+            {
+                {"wrnamespace", NamespaceIds == null ? null : MediaWikiHelper.JoinValues(NamespaceIds)},
+                {"wrlimit", PaginationSize},
+                {"wrprop", ShowChangedTime ? "changed" : null},
```

Comment:
1. I think you can always enable `ShowChangedTime`, if you need it.
2. Since there is custom field (`ChangedTime`), you may need to set up a wiki list item type (like `GeoSearchResultItem`) and override the `WikiPageGenerator<TItem>.ItemFromJson` method, so that `WikiPageGenerator<TItem>.EnumItemsAsync` can enumerate something containing a `PageStub` and a `ChangedTime`, rather than `PageStub` only.

Reply:
@CXuesong. I don't like that I hardcoded the value in FindQueryResponseItemsRoot but I couldn't find other lists that give results in a different format (I tested on Armenian Wikipedia).

Reply:
Thanks!

> I don't like that I hardcoded the value in FindQueryResponseItemsRoot but I couldn't find other lists that give results in a different format (I tested on Armenian Wikipedia).

I guess this is the first time I discover a query result without `query` JSON root node. Yeah probably we can go without any generic root-finding-function for now, and see what happens.